src/nxt_conf_validation.c

```diff
@@ -1936,10 +1932,13 @@ static nxt_int_t
 nxt_conf_vldt_proxy(nxt_conf_validation_t *vldt, nxt_conf_value_t *value,
     void *data)
 {
-    nxt_str_t       name;
+    nxt_str_t       name, *ret;
     nxt_sockaddr_t  *sa;
 
-    nxt_conf_get_string(value, &name);
+    ret = nxt_conf_get_string_dup(value, vldt->pool, &name);
+    if (nxt_slow_path(ret == NULL)) {
+        return NXT_ERROR;
+    }
 
```

Comment:
I take it this doesn't get called from nxt_conf_vldt_object_iterator()?

Reply:
Right.

Both `"listeners"` and `"servers"` have the Unix socket in the object name, which results in a validator stored in `.u.object`, buit `"proxy"` has the Unix socket in the string, and so it results in a validator stored in `.validator`.

(recall from yesterday:

```c
$ grep -rn '=.nxt_conf_vldt_\(listener\|proxy\|server\)\>'
src/nxt_conf_validation.c:280:        .u.object   = nxt_conf_vldt_listener,
src/nxt_conf_validation.c:781:        .validator  = nxt_conf_vldt_proxy,
src/nxt_conf_validation.c:1350:        .u.object   = nxt_conf_vldt_server,
```
)

`.u.object` is a `nxt_conf_vldt_member_t`, which I found to be called here:

```c
$ grepc -tuf nxt_conf_vldt_member_t . | grep -e ^nxt_ -e '^[{}]' -e validator -e name
nxt_conf_vldt_object_iterator(nxt_conf_validation_t *vldt,
{
    nxt_str_t               name, str;
    nxt_conf_vldt_member_t  validator;
    validator = (nxt_conf_vldt_member_t) data;
        member = nxt_conf_next_object_member(value, &name, &index);
        if (nxt_slow_path(nxt_str_dup(vldt->pool, &str, &name) == NULL)) {
        ret = validator(vldt, &str, member);
}
```

`.validator` I found it to be called here:

```c
$ grepc -tuf '>validator\(' . | grep -e ^nxt_ -e '^[{}]' -e '>validator('
nxt_conf_vldt_object(nxt_conf_validation_t *vldt, nxt_conf_value_t *value,
{
                ret = vals->validator(vldt, member, vals->u.members);
}
```

`.validator()` isn't being passed the name, which is why it needs to get it inside the call.

Reply:
Context:

```diff
@@ -1936,10 +1932,13 @@ static nxt_int_t
 nxt_conf_vldt_proxy(nxt_conf_validation_t *vldt, nxt_conf_value_t *value,
```